tests/components/dyson/test_fan.py

```diff
@@ -320,14 +399,354 @@ def test_service_set_night_mode(self):
         self.hass.data[DYSON_DEVICES] = []
         dyson_device.entity_id = 'fan.living_room'
         self.hass.data[dyson.DYSON_FAN_DEVICES] = [dyson_device]
-        dyson.setup_platform(self.hass, None, mock.MagicMock())
+        dyson.setup_platform(self.hass, None,
+                             mock.MagicMock(), mock.MagicMock())
 
-        self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
+        self.hass.services.call(dyson.DYSON_DOMAIN,
+                                dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.bed_room",
                                  "night_mode": True}, True)
-        assert not dyson_device.night_mode.called
+        assert not dyson_device.set_night_mode.called
 
-        self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
+        self.hass.services.call(dyson.DYSON_DOMAIN,
+                                dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.living_room",
                                  "night_mode": True}, True)
-        dyson_device.night_mode.assert_called_with(True)
+        dyson_device.set_night_mode.assert_called_with(True)
+
+
+@asynctest.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+@asynctest.patch('libpurecool.dyson.DysonAccount.devices',
+                 return_value=[_get_dyson_purecool_device()])
+async def test_purecool_turn_on(devices, login, hass):
+    """Test turn on."""
+    device = devices.return_value[0]
+    await async_setup_component(hass, dyson.DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(DOMAIN, SERVICE_TURN_ON,
+                                   {ATTR_ENTITY_ID: "fan.bed_room"}, True)
+    assert not device.turn_on.called
```

Comment:
```py
assert  len(device.turn_on.call_count) == 0
```

Reply:
Done. Thanks for the tip.

Reply:
`call_count` returns an integer, not a list as I wrote above. Sorry about that.